src/lib.rs

```diff
@@ -234,10 +241,35 @@ impl Default for ConnectionOptions<options_typestate::NoAuth> {
             max_reconnects: Some(60),
             disconnect_callback: Callback(None),
             reconnect_callback: Callback(None),
+            tls_connector: None,
+            tls_required: false,
         }
     }
 }
 
+impl<T> fmt::Debug for ConnectionOptions<T> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> Result<(), fmt::Error> {
+        f.debug_map()
+            .entry(&"auth", &self.auth)
+            .entry(&"name", &self.name)
+            .entry(&"no_echo", &self.no_echo)
+            .entry(&"reconnect_buffer_size", &self.reconnect_buffer_size)
+            .entry(&"max_reconnects", &self.max_reconnects)
+            .entry(&"disconnect_callback", &self.disconnect_callback)
+            .entry(&"reconnect_callback", &self.reconnect_callback)
+            .entry(
+                &"tls_connector",
+                if self.tls_connector.is_some() {
+                    &"set"
+                } else {
+                    &"unset"
+                },
+            )
+            .entry(&"tls_required", &self.tls_required)
+            .finish()
+    }
+}
```

Comment:
This had to be manually implemented because `native_tls::TlsConnector` unfortunately does not implement it.

Reply:
Upstream PR maybe?

Reply:
I see there's been an outstanding issue since 2018, I'll give it a peek and see if it's fairly trivial https://github.com/sfackler/rust-native-tls/issues/99

Reply:
If its easy